src/FuncHW/ListExtensions.cs

```diff
@@ -12,5 +12,17 @@ public static class ListExtensions
         //add extension for List CountElements: receives Func, returns the number of elements that match the condition
 
         //add extension for List SelectWhereNot: receives Func, returns the list of elements that don't match the condition
+
+        public static T GetLast<T>(this List<T> values, Func<T, bool> func) where T : new()
+        {
+            for (int i = values.Count-1; i >= 0; i--)
+            {
+                if (func(values[i]))
+                {
+                    return values[i];
+                }
+            }
+            return new T();
```

Comment:
it's good, you thinking about case when there would not be values matching condition.
but task is to return first value that matchs' condition; so if there is not such values, return null
to return null nedeed to add where T : class (struct can't be null)